### DISCUSSION TITLE
Unable to change icon of a CTkToplevel widget

### DISCUSSION THREAD

**Discussion by NanoParticles** [TomSchimansky/CustomTkinter#1968] [Q&A]
Im unable to change the icon of a CTkToplevel widget. I tried using iconbitmap, but doesnt work. Is there any alternative methods that I can use?

**Comment by Akascape** [comment]
**Try this method:**


```python
import os
from PIL import ImageTk
...

self.iconpath = ImageTk.PhotoImage(file=os.path.join("assets","logo.png"))

self.wm_iconbitmap()
self.after(300, lambda: self.iconphoto(False, self.iconpath))
```

**Reply by NanoParticles** [reply]
doesnt it need to be a ico file or?

and besides, it come back with a error:

Traceback (most recent call last):
  File "c:\Users\mikeh\OneDrive\Desktop\python builds\Minecraft Server Crafter\Releases\Lite\0.0.15\main.py", line 255, in attachServerJar
    ServerJarSelection = FileSelectWindow(root)
                         ^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\mikeh\OneDrive\Desktop\python builds\Minecraft Server Crafter\Releases\Lite\0.0.15\altdialog.py", line 93, in __init__
    self.after(300,lambda: self.iconphoto(False,self.iconpath))
  File "C:\Program Files\Python311\Lib\tkinter\__init__.py", line 872, in after
    name = self._register(callit)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Python311\Lib\tkinter\__init__.py", line 1584, in _register
    self.tk.createcommand(name, f)
    ^^^^^^^
AttributeError: 'FileSelectWindow' object has no attribute 'tk'

It works when I comment out the lines you gave

**Reply by Akascape** [reply]
@NanoParticles iconbitmap is used for .ico file, which doesn't work properly on all platform (like mac). But iconphoto method is better and works in all platforms properly, and you have to use .png file.

**Reply by NanoParticles** [reply]
Well, it doesnt work for me, so I might leave it as is since its a whole can of worms too big to open xD. I used my folder structure and everything, in place of the os.path.join() statement. This is what I used:

```
self.rootDir = ""
(self.rootDir is updated outside of the class and sets the root directory of my project into the variable)
self.iconpath = ImageTk.PhotoImage(file=os.path.join(str(self.rootDir),"base/ui/default.png"))
self.wm_iconbitmap()
self.after(300, lambda: self.iconphoto(False, self.iconpath))
```
Am I referencing this right? Am I forgetting something?

Does your answer work for you @Akascape ?
